executor/common_linux.h

```diff
@@ -1734,6 +1732,141 @@ static int do_sandbox_namespace(void)
 }
 #endif
 
+#if SYZ_EXECUTOR || SYZ_SANDBOX_ANDROID_UNTRUSTED_APP
+#include <fcntl.h> // open(2)
+#include <grp.h> // setgroups
+#include <sys/xattr.h> // setxattr, getxattr
+
+#define AID_NET_BT_ADMIN 3001
+#define AID_NET_BT 3002
+#define AID_INET 3003
+#define AID_EVERYBODY 9997
+#define AID_APP 10000
+
+#define UNTRUSTED_APP_UID AID_APP + 999
+#define UNTRUSTED_APP_GID AID_APP + 999
+
+const char* SELINUX_CONTEXT_UNTRUSTED_APP = "u:r:untrusted_app:s0:c512,c768";
+const char* SELINUX_LABEL_APP_DATA_FILE = "u:object_r:app_data_file:s0:c512,c768";
+const char* SELINUX_CONTEXT_FILE = "/proc/thread-self/attr/current";
+const char* SELINUX_XATTR_NAME = "security.selinux";
+
+gid_t UNTRUSTED_APP_GROUPS[] = {UNTRUSTED_APP_GID, AID_NET_BT_ADMIN, AID_NET_BT, AID_INET, AID_EVERYBODY};
+size_t UNTRUSTED_APP_NUM_GROUPS = sizeof(UNTRUSTED_APP_GROUPS) / sizeof(UNTRUSTED_APP_GROUPS[0]);
+
+// Similar to libselinux getcon(3), but withouthe library dependency
+// and without any dynamic memory allocation.
+static int getcon(char* context, size_t context_size)
+{
+	int fd = open(SELINUX_CONTEXT_FILE, O_RDONLY);
+
+	if (fd < 0)
+		fail("getcon: Couldn't open %s", SELINUX_CONTEXT_FILE);
+
+	ssize_t nread = read(fd, context, context_size);
+
+	close(fd);
+
+	if (nread < 0)
+		fail("getcon: Failed to read from %s", SELINUX_CONTEXT_FILE);
+
+	// The contents of the context file MAY end with a newline
+	// and MAY not have a null terminator.  Handle this here.
+	if (context[nread - 1] == '\n')
+		context[nread - 1] = '\0';
+
+	return 0;
+}
+
+// see: man 3 setcon
+static int setcon(const char* context)
+{
+	char new_context[512];
+
+	// Attempt to write the new context
+	int fd = open(SELINUX_CONTEXT_FILE, O_WRONLY);
+
+	if (fd < 0)
+		fail("setcon: Could not open %s", SELINUX_CONTEXT_FILE);
+
+	ssize_t bytes_written = write(fd, context, strlen(context));
+	close(fd);
+
+	if (bytes_written != (ssize_t)strlen(context))
+		fail("setcon: Could not write entire context.  Wrote %zi, expected %zu", bytes_written, strlen(context));
+
+	// Validate the transition by checking the context
+	if (getcon(new_context, sizeof(new_context) != 0))
+		fail("setcon: Could not read context");
+
+	if (strcmp(context, new_context))
+		fail("setcon: Failed to change to %s, context is %s", context, new_context);
+
+	return 0;
+}
+
+// Similar to libselinux getfilecon(3), but without the library dependency
+// and without any dynamic memory allocation.
+static int getfilecon(const char* path, char* context, size_t context_size)
+{
+	if (getxattr(path, SELINUX_XATTR_NAME, context, context_size) != 0)
+		fail("getfilecon: getxattr failed");
+
+	return 0;
+}
+
+// see: man 3 setfilecon
+static int setfilecon(const char* path, const char* context)
+{
+	char new_context[512];
+	int retval = setxattr(path, SELINUX_XATTR_NAME, context, strlen(context) + 1, 0);
```

Comment:
In such cases we generally do (saves 1 line):
```
	if (setxattr(path, SELINUX_XATTR_NAME, context, strlen(context) + 1, 0))
		fail("setfilecon: setxattr failed");
```
But I am confused about return values here too. man says these functions return size:
http://man7.org/linux/man-pages/man2/getxattr.2.html

Reply:
Lines aren't expensive, it seems silly to pursue code density for the sake of itself.

`setxattr` returns zero on success.

Reply:
> setxattr returns zero on success.

doh broken man pages

Reply:
> Lines aren't expensive, it seems silly to pursue code density for the sake of itself.

Lines are exactly what everybody needs to read, understand and maintain.
So instead of:
```
int b = a;
b += 1;
int c = f(b);
int d = c - 1;
return d;
```
I would prefer:
```
return f(a + 1) - 1;
```
In the above case, splitting code into simpler pieces isn't just not making it simpler to understand as a whole, but actually making it harder to understand as a whole.

While complex splitting logic into more lines/functions can make it simpler to understand, introducing additional lines/functions all else being equal increases maintenance cost.

This also has to do with project conventions. Throughout the executor we do `if (foo(...)) fail`. So once a reader gets used to that style, parsing another instance does not produce additional cognitive load. While diverging from the common style does produce additional cognitive load while reading -- are we doing something different here? will we use the return value for anything other than checking for error and bailing out?

Reply:
I think this isn't a complex situation like the example, but I can put everything on one line if that's the preferred style.

Reply:
This is the _existing_ style. Try `grep -B 1 "fail(" executor/*.{cc,h}`.